src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,552 @@
+				case gl::INTERNAL_RGBA32I: return getTranslatedFinalFormat(EF_R32G32B32A32_SINT);			//GL_RGBA32I
+
+				// Floating formats
+				case gl::INTERNAL_R16F: return getTranslatedFinalFormat(EF_R16_SFLOAT);				//GL_R16F
+				case gl::INTERNAL_RG16F: return getTranslatedFinalFormat(EF_R16G16_SFLOAT);			//GL_RG16F
+				///case gl::INTERNAL_RGB16F: return getTranslatedFinalFormat(EF_R16G16B16_SFLOAT);			//GL_RGB16F
+				case gl::INTERNAL_RGBA16F: return getTranslatedFinalFormat(EF_R16G16B16A16_SFLOAT);			//GL_RGBA16F
+
+				case gl::INTERNAL_R32F: return getTranslatedFinalFormat(EF_R32_SFLOAT);				//GL_R32F
+				case gl::INTERNAL_RG32F: return getTranslatedFinalFormat(EF_R32G32_SFLOAT);			//GL_RG32F
+				///case gl::INTERNAL_RGB32F: return getTranslatedFinalFormat(EF_R32G32B32_SFLOAT);			//GL_RGB32F
+				case gl::INTERNAL_RGBA32F: return getTranslatedFinalFormat(EF_R32G32B32A32_SFLOAT);			//GL_RGBA32F
+
+				case gl::INTERNAL_R64F_EXT: return getTranslatedFinalFormat(EF_R64_SFLOAT);			//GL_R64F
+				case gl::INTERNAL_RG64F_EXT: return getTranslatedFinalFormat(EF_R64G64_SFLOAT);		//GL_RG64F
+				///case gl::INTERNAL_RGB64F_EXT: return getTranslatedFinalFormat(EF_R64G64B64_SFLOAT);		//GL_RGB64F
+				case gl::INTERNAL_RGBA64F_EXT: return getTranslatedFinalFormat(EF_R64G64B64A64_SFLOAT);		//GL_RGBA64F
+
+				// sRGB formats
+				case gl::INTERNAL_SR8: return getTranslatedFinalFormat(EF_R8_SRGB);				//GL_SR8_EXT
+				case gl::INTERNAL_SRG8: return getTranslatedFinalFormat(EF_R8G8_SRGB);				//GL_SRG8_EXT
+				case gl::INTERNAL_SRGB8: return getTranslatedFinalFormat(EF_R8G8B8_SRGB);			//GL_SRGB8
+				case gl::INTERNAL_SRGB8_ALPHA8: return getTranslatedFinalFormat(EF_R8G8B8A8_SRGB);		//GL_SRGB8_ALPHA8
+
+				// Packed formats
+				case gl::INTERNAL_RGB9E5: return getTranslatedFinalFormat();			//GL_RGB9_E5
+				case gl::INTERNAL_RG11B10F: return getTranslatedFinalFormat();			//GL_R11F_G11F_B10F
+				case gl::INTERNAL_RG3B2: return getTranslatedFinalFormat();			//GL_R3_G3_B2
+				case gl::INTERNAL_R5G6B5: return getTranslatedFinalFormat();			//GL_RGB565
+				case gl::INTERNAL_RGB5A1: return getTranslatedFinalFormat();			//GL_RGB5_A1
+				case gl::INTERNAL_RGBA4: return getTranslatedFinalFormat();			//GL_RGBA4
+
+				case gl::INTERNAL_RG4_EXT: return getTranslatedFinalFormat();
+
+				// Luminance Alpha formats
+				case gl::INTERNAL_LA4: return getTranslatedFinalFormat();				//GL_LUMINANCE4_ALPHA4
+				case gl::INTERNAL_L8: return getTranslatedFinalFormat();				//GL_LUMINANCE8
+				case gl::INTERNAL_A8: return getTranslatedFinalFormat();				//GL_ALPHA8
+				case gl::INTERNAL_LA8: return getTranslatedFinalFormat();				//GL_LUMINANCE8_ALPHA8
+				case gl::INTERNAL_L16: return getTranslatedFinalFormat();				//GL_LUMINANCE16
+				case gl::INTERNAL_A16: return getTranslatedFinalFormat();				//GL_ALPHA16
+				case gl::INTERNAL_LA16: return getTranslatedFinalFormat();				//GL_LUMINANCE16_ALPHA16
```

Comment:
these can be expanded to RGB8_SRGB and RGBA8_SRGB

Reply:
but what about `LA4`?

Reply:
there is no enough SRGB `EF_FORMAT`s to cover it

Reply:
LA4 is 4 bits, just expand the format into 8bits per channel and pretend the format is RGBA8_SRGB.

Reply:
to fix in next commit in that case

Reply:
will `EF_R4G4B4A4_UNORM_PACK16` fit it?

Reply:
not really, because for every two LA4  pixels you'd be getting one `EF_R4G4B4A4_UNORM_PACK16` pixel

Reply:
so padding needed.

Reply:
also SRGB concerns